components/waitwhile/actions/add-customer-note-entry/add-customer-note-entry.ts

```diff
@@ -0,0 +1,52 @@
+import waitwhile from "../../app/waitwhile.app";
+import { defineAction } from "@pipedream/types";
+
+export default defineAction({
+  name: "Add Customer Note Entry",
+  version: "0.0.1",
+  key: "add-customer-note-entry",
+  description: "Add a customer note entry",
+  props: {
+    waitwhile,
+    customerId: {
+      label: "CustomerID",
+      type: "string",
+      description: "Identifier of customer, automatically derived from visitor contact information if not provided."
+    },
```

Comment:
It would be great if you can put the `customerId` prop in the app file (waitwhile.app.ts) and use [async options](https://pipedream.com/docs/components/migrating/#advanced-using-async-options) to list the customers. Then the prop can be reused by other components. Take a look at some of the Trello props as an example: https://github.com/PipedreamHQ/pipedream/blob/master/components/trello/trello.app.js